Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,278 @@
+    }
+
+    // MARK: UX
+    
+    private struct UX {
+        static let defaultInset = UIEdgeInsets(equalInset: 19)
+        static let contentSizeChange: CGFloat = 64
+    }
+
+    // MARK: Properties
+    
+    private let theme: Theme
+
+    private let contentStackView = UIStackView().then {
+        $0.axis = .vertical
+        $0.spacing = 3
+        $0.alignment = .fill
+        $0.layoutMargins = UX.defaultInset
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+
+    var actionHandler: ((Action) -> Void)?
+
+    // MARK: Lifecycle
+    
+    init(theme: Theme) {
+        self.theme = theme
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.Share.shareScreenTitle
+
+        doLayoutAndAddGestureRecognizers()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        preferredContentSize = contentStackView.systemLayoutSizeFitting(
+            CGSize(width: view.bounds.size.width - UX.contentSizeChange,
+                   height: view.bounds.size.height - UX.contentSizeChange),
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        ).with {
+            $0.width += UX.contentSizeChange
+        }
+
+        navigationController?.preferredContentSize = preferredContentSize
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(theme)
+            }
+        }
+    }
+
+    // MARK: Internal
+    
+    private func doLayoutAndAddGestureRecognizers() {
+        let shareTypeViewEmail = ShareTypeView(type: .email)
+        let shareEmailGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewEmail.addGestureRecognizer(shareEmailGesture)
+
+        let shareTypeViewTwitter = ShareTypeView(type: .twitter)
+        let shareTwitterGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewTwitter.addGestureRecognizer(shareTwitterGesture)
+
+        let shareTypeViewFacebook = ShareTypeView(type: .facebook)
+        let shareFacebookGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewFacebook.addGestureRecognizer(shareFacebookGesture)
+
+        let shareTypeViewDefault = ShareTypeView(type: .default)
+        let shareDefaultGesture = UITapGestureRecognizer(target: self, action: #selector(shareClicked(_:)))
+        shareTypeViewDefault.addGestureRecognizer(shareDefaultGesture)
```

Comment:
Is there a reason we're using tap gestures to handle actions here instead of `ShareTypeView` being a `UIControl` subclass?

Reply:
No reason, I try not to make views UIcontrol but in this case it is necessary, I agree.